sonic-thermalctld/scripts/thermalctld

```diff
@@ -0,0 +1,535 @@
+
+        self.under_temper = status
+        return True
+
+
+#
+# TemperUpdater ======================================================================
+#
+class TemperUpdater(object):
+    # Temperature information table name in database
+    TEMPER_INFO_TABLE_NAME = 'TEMPERATURE_INFO'
+
+    def __init__(self, chassis):
+        """
+        Constructor of TemperUpdater
+        :param chassis: Object representing a platform chassis
+        """
+        self.chassis = chassis
+        self.temper_status_dict = {}
+        state_db = daemon_base.db_connect(swsscommon.STATE_DB)
+        self.table = swsscommon.Table(state_db, TemperUpdater.TEMPER_INFO_TABLE_NAME)
+
+    def destroy(self):
+        """
+        Destructor of TemperUpdater
+        :return:
+        """
+        for name in self.temper_status_dict.keys():
+            self.table._del(name)
+
+    def update(self):
+        """
+        Update all temperature information to database
+        :return:
+        """
+        logger.log_info("Start temperature updating")
+        for index, thermal in enumerate(self.chassis.get_all_thermals()):
+            self._refresh_temper_status(thermal, index)
+
+        logger.log_info("Stop temperature updating")
+
+    def _refresh_temper_status(self, thermal, index):
+        """
+        Get temperature status by platform API and write to database
+        :param thermal: Object representing a platform thermal zone
+        :param index: Index of the thermal object in platform chassis
+        :return:
+        """
+        name = try_get(thermal.get_name, 'Thermal {}'.format(index + 1))
+        if name not in self.temper_status_dict:
+            self.temper_status_dict = TemperStatus()
+
+        temper_status = self.temper_status_dict[name]
+
+        temperature = try_get(thermal.get_temperature)
+        high_threshold = try_get(thermal.get_high_threshold)
+        low_threshold = try_get(thermal.get_low_threshold)
+
+        warning = False
+        if temper_status.set_over_temper(temperature, high_threshold):
+            self._on_over_temper_status_changed(name, temperature, high_threshold, temper_status)
+            warning = True
+
+        if temper_status.set_under_temper(temperature, low_threshold):
+            self._on_under_temper_status_changed(name, temperature, high_threshold, temper_status)
+            warning = True
+
+        fvs = swsscommon.FieldValuePairs(
+            [('temperature', str(temperature)),
+             ('high_threshold', str(high_threshold)),
+             ('low_threshold', str(low_threshold)),
+             ('warning_status', str(warning)),
+             ('critical_high_threshold', NOT_AVAILABLE),
+             ('critical_low_threshold', NOT_AVAILABLE),
+             ('timestamp', datetime.now().strftime('%Y%m%d %H:%M:%S'))
+             ])
+
+        self.table.set(name, fvs)
+
+    def _on_over_temper_status_changed(self, name, temperature, high_threshold, temper_status):
```

Comment:
suggest extracting the common logic in _on_over_temper_status_changed and _on_under_temper_status_changed into a function.